### ISSUE TITLE
Add option to delete stack on pulumi destroy

### ISSUE THREAD

**Issue by nicholasgibson2** [pulumi/actions#681]
@Moon1706 proposed this feature in https://github.com/pulumi/actions/pull/584, bulk of the work was already completed as part of the original PR. This feature would be very useful when dealing with ephemeral stacks, and prevent needing an additional cleanup step. Related issue https://github.com/pulumi/actions/issues/41

**Comment by nicholasgibson2** [comment]
> Suppose we supported the `pulumi stack` command. Would that satisfy your use case?

@RobbieMcKinstry yeah this would work, and having a dedicated `pulumi stack` command like this would probably be the best approach since it keeps parity with the CLI (although I would personally prefer being able to do everything in one single command).

> there's no precedence in the CLI for removing the stack when destroy is run

I would actually love to have this functionality in the CLI as well 😄

> Is this something you'd like to contribute?

Don't have capacity to commit to this right now, I'll keep you posted if that changes tho.

**Comment by dixler** [comment]
https://github.com/pulumi/pulumi/issues/10484 addresses this FYI
